pkg/controller/volume/attachdetach/cache/actual_state_of_world.go

```diff
@@ -433,21 +431,26 @@ func (asw *actualStateOfWorld) addVolumeToReportAsAttached(
 	}
 }
 
-func (asw *actualStateOfWorld) ResetNodeStatusUpdateNeeded(
-	nodeName string) error {
-	asw.Lock()
-	defer asw.Unlock()
-	// Remove volume from volumes to report as attached
+// Update the flag statusUpdateNeeded to indicate whether node status is already updated or
+// needs to be updated again by the node status updater.
+// This is an internal function and caller should acquire and release the lock
+func (asw *actualStateOfWorld) setNodeStatusUpdateNeeded(nodeName string, needed bool) {
```

Comment:
nit: to avoid confusing this with the `SetNodeStatusUpdateNeeded` which sets the value to `true`, maybe rename it to `modifyNodeStatusUpdateNeeded`?